SLE Release Notes 2025-02 (#246)
ni/install-systemlink-enterprise: release-notes/2025-02/README.md

# SystemLink Enterprise release year-release-month Release Notes

The 2025-02 release for SystemLink Enterprise has been published to
<https://downloads.artifacts.ni.com>. This update includes new features, bug
fixes, and security updates. Work with your account representative to obtain
credentials to access these artifacts. If you are not upgrading from the
previous release, refer to past release notes to ensure you have addressed all
required configuration changes.

## Upgrading from the release year-release-month to the release year-release-month

<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->

> ## New Features and Behavior changes

Reviewer: Include info for February release at https://ni-staging.zoominsoftware.io/docs/en-US/bundle/systemlink-enterprise/page/new-features-and-changes.html

- Behavior change or new feature description

- Behavior change or new feature description

> ## Helm Chart Breaking Changes

Reviewer: Include information provided in https://emerson.sharepoint.com/:x:/s/ASWSystemLink/EfjxeCtltptCp5CCWBVdxXIBjV8dhEvqeoMR6PPcqFkw3A?e=mOQIwU

Reviewer 2: @prestwick , I did include the information documented by the squads in 
- [SLE 2025-02_Helm Chart Breaking Changes, Software Behavior Changes (including Privileges) for the Release Notes - out of January 2025.xlsx](https://emerson.sharepoint.com/:x:/s/ASWSystemLink/EfjxeCtltptCp5CCWBVdxXIBjV8dhEvqeoMR6PPcqFkw3A?e=YrlGXH)

The last 2x bullets for Behavior Changes for DataFrameService seem to be internal documentation only.

# SystemLink Enterprise 2025-02 Release Notes

The 2025-02 release for SystemLink Enterprise has been published to
<https://downloads.artifacts.ni.com>. This update includes new features, bug
fixes, and security updates. Work with your account representative to obtain
credentials to access these artifacts. If you are not upgrading from the
previous release, refer to past release notes to ensure you have addressed all
required configuration changes.

## Upgrading from the 2025-01 to the 2025-02

<!-- Optional section to include comments and instructions needed to successfully upgrade from the previous release to the current release. If the only changes needed are already captured in Helm Chart Breaking Changes, this section is not needed. -->

## New Features and Behavior changes

- `dataframeservice:1.14.61`

>   - The service now limits the length of strings that appear in table and column "properties" dictionaries. Property keys can be up to 256 characters, and values can be up to 1000 characters.

Reviewer 3: ```suggestion
  - The service now limits the length of strings that appear in table and column "properties" dictionaries. The limit for property keys is 256 characters. The limit for values is 1000 characters.
```

- Behavior change or new feature description

## Helm Chart Breaking Changes

> - New required secret `dataframeservice.secrets.nessie.bearerToken`

Reviewer 2: This was documented by by Chris Pawling in
- [SLE 2025-02_Helm Chart Breaking Changes, Software Behavior Changes (including Privileges) for the Release Notes - out of January 2025.xlsx](https://emerson.sharepoint.com/:x:/s/ASWSystemLink/EfjxeCtltptCp5CCWBVdxXIBjV8dhEvqeoMR6PPcqFkw3A?e=RN9RYY)

And I found this PR that included the changes to the secrets and values yaml files
- [Add nessie credential secret to values template #245](https://github.com/ni/install-systemlink-enterprise/pull/245/files)

  - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2025-02/getting-started/templates/systemlink-values.yaml#L735)
  - [View this secret configuration](https://github.com/ni/install-systemlink-enterprise/blob/2025-02/getting-started/templates/systemlink-secrets.yaml#L252)

> - Dremio was upgraded and it will be necessary to be reset after the upgrade.

Reviewer 2: Documented by Phil Vallone in
- [SLE 2025-02_Helm Chart Breaking Changes, Software Behavior Changes (including Privileges) for the Release Notes - out of January 2025.xlsx](https://emerson.sharepoint.com/:x:/s/ASWSystemLink/EfjxeCtltptCp5CCWBVdxXIBjV8dhEvqeoMR6PPcqFkw3A?e=RN9RYY)

>   - Versions of the dataframeservice prior to 0.8 created an index in MongoDB that is no longer being used. Starting in 0.8, the DataFrame Service automatically dropped the unused index. However, starting with 1.14.52, we no longer drop the index. When upgrading directly from 0.8 or earlier to 1.14.52 or later, it is recommended (but not required) to manually drop the index named "SupportsAppend_1" from the "tables" collection of the "nidataframe" database.

Reviewer 3: ```suggestion
  - DataFrame Service creates an index called "SupportsAppend_1" in MongoDB. Starting with version 0.8, the DataFrame Service automatically dropped the "SupportsAppend_1" index. Starting with version 1.14.52, the DataFrame Service no longer drop the"SupportsAppend_1" index. When upgrading directly from version 0.8 or earlier to version 1.14.52 or later, we recommend that you drop the "SupportsAppend_1" index from the "tables" collection of the "nidataframe" database manually.
```

>   - Dremio was upgraded to 25.2.2 to pull in fixes for critical vulnerabilities and functionality necessary for data table compaction. Upgrading will require the customer to reset Dremio: [https://dev.azure.com/ni/DevCentral/_wiki/wikis/Stratus/83721/How-to-reset-Dremio](https://dev.azure.com/ni/DevCentral/_wiki/wikis/Stratus/83721/How-to-reset-Dremio). Customer success should be familiar with procedure so should be capable of guiding customers through the upgrade. The Dremio reset procedure does not result in any loss of data.

Reviewer 3: ```suggestion
  - Dremio was upgraded to 25.2.2. The update includes fixes for critical vulnerabilities and added functionality for data table compaction. You must reset Dremio for this upgrade. Refer to Resetting Dremio, [https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/resetting-dremio.html](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/resetting-dremio.html) in the SystemLink Enterprise User Manual. Resetting Dremio does not result in loss of data.
```

>   - Data Table compaction (utilizing Iceberg) is now turned on by default. This introduces a new replica set of pods associated with the DFS "dataframeservice-nessie-*". This affects the resource footprint of Dremio, with the default total executor sizing going from ~60 CPU/~500GB memory to ~80 CPU/~350GB memory. The following PR can be referenced for exact numbers: [https://ni.visualstudio.com/DevCentral/_git/Skyline/commit/c1055fea5d38c33c7c4c513e0295e7f5e6a1bc95?refName=refs%2Fheads%2Fmaster](https://ni.visualstudio.com/DevCentral/_git/Skyline/commit/c1055fea5d38c33c7c4c513e0295e7f5e6a1bc95?refName=refs%2Fheads%2Fmaster)

Reviewer 3: ```suggestion
  - Data table compaction is enabled by default. Data table compaction uses Iceberg and introduces a new replica set of pods associated with the DataFrame Service "dataframeservice-nessie-*". The replica set of pods affects the resource footprint of Dremio, The default total executor sizing increases from ~60 CPU/~500GB memory to ~80 CPU/~350GB memory. Refer to the following PRfor exact numbers: [https://ni.visualstudio.com/DevCentral/_git/Skyline/commit/c1055fea5d38c33c7c4c513e0295e7f5e6a1bc95?refName=refs%2Fheads%2Fmaster](https://ni.visualstudio.com/DevCentral/_git/Skyline/commit/c1055fea5d38c33c7c4c513e0295e7f5e6a1bc95?refName=refs%2Fheads%2Fmaster)
```

>     - [DFS: Update pilot config for the compaction feature #232](https://github.com/ni/install-systemlink-enterprise/pull/232/files)

Reviewer 3: ```suggestion
    - [DataFrame Service: Update pilot config for the compaction feature #232](https://github.com/ni/install-systemlink-enterprise/pull/232/files)
```

  - Dremio was upgraded and it will be necessary to be reset after the upgrade.
    - Follow the instructions in [Resetting Dremio](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/resetting-dremio.html)

## Upgrade Considerations

### RabbitMQ Version

SystemLink Enterprise includes a deployment of the
[RabbitMQ](https://www.rabbitmq.com/) message bus. Since you cannot skip minor
versions when updating RabbitMQ, you may not be able to upgrade directly between
versions of SystemLink Enterprise. The table below shows the version of the
RabbitMQ dependency for each released version of SystemLink Enterprise. Refer to
[Updating SystemLink Enterprise](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/updating-systemlink-enterprise.html)
for detailed update instructions.

| RabbitMQ Version | First SystemLink Enterprise Version | Last SystemLink Enterprise Version |
| ---------------- | ----------------------------------- | ---------------------------------- |
| 3.11.x           | 0.12.x                              | 0.15.x                             |
| 3.12.x           | 0.16.x                              | 0.24.x                             |

> | 3.13.x           | 0.25.x                              | current                            |

Reviewer 2: @prestwick , in [the 2025-01 Readme](https://github.com/ni/install-systemlink-enterprise/tree/main/release-notes/2025-01) the RabbitMQ version does also show version 4.0.x, and I recalled the template was updated to include this version but I no longer see this change in the template.
